sofl/utils/archive_utils.py

```
@@ -0,0 +1,123 @@
+                info_list = rf.infolist()
+
+                if not info_list:
+                    logging.warning("Archive is empty")
+                    return False
+
+                # Проверяем пароль, пытаясь прочитать первый файл
+                try:
+                    with rf.open(info_list[0]) as f:
+                        f.read(1)  # Читаем хотя бы 1 байт
+                    logging.info("Archive verified successfully via rarfile")
+                    return True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)